plaso/parsers/dsv_parser.py

```diff
@@ -114,6 +118,63 @@ def _CreateDictReader(self, line_reader):
         quotechar=quotechar, restkey=magic_test_string,
         restval=magic_test_string)
 
+  # pylint: disable=missing-return-type-doc
+  def _CreateLineReader(self, file_object):
+    """Returns an object that returns lines from a text file.
+
+    The line reader is advanced to the beginning of the DSV content, skipping
+    any header lines.
+
+    Args:
+      file_object (dfvfs.FileIO): file-like object.
+
+    Returns:
+      TextFile|BinaryLineReader: an object that implements an iterator
+          over lines in a text file.
+
+    Raises:
+      UnicodeDecodeError: if the file cannot be read with the specified
+          encoding.
+    """
+    # The Python 2 csv module reads bytes and the Python 3 csv module Unicode
+    # reads strings.
+    if py2to3.PY_3:
+      line_reader = text_file.TextFile(file_object, encoding=self._encoding)
+    else:
+      line_reader = line_reader_file.BinaryLineReader(file_object)
+    # If we specifically define a number of lines we should skip, do that here.
+    for _ in range(0, self.NUMBER_OF_HEADER_LINES):
+      try:
+        line_reader.readline(self._maximum_line_length)
+      except UnicodeDecodeError:
+        raise
+    return line_reader
+
+  def _HasExpectedLineLength(self, file_object):
+    """Determines if a file begins with lines of the expected length.
+
+    As we know the maximum length of valid lines in the DSV file, the presence
+    of lines longer than this indicates that the file will not be parsed
+    successfully, without reading excessive data from a large file.
+
+    Args:
+      file_object (dfvfs.FileIO): file-like object.
+
+    Returns:
+        bool: True if the file has lines of the expected length.
+    """
+    original_file_position = file_object.tell()
+    line_reader = self._CreateLineReader(file_object)
+    for _ in range(0, 20):
+      # Attempt to read a line that twice as long as any line that should be in
+      # the file.
+      sample_line = line_reader.readline(self._maximum_line_length * 2)
```

Comment:
shouldn't self._maximum_line_length + 1 be enough of a signal?

Reply:
Good point - I was thinking of trying to optimize caching here, but both line reader use their own read caches. Changed this.